### ISSUE TITLE
[BUG] Disease/Flu transfers to new builder

### ISSUE THREAD

**Issue by Cursedth** [ldtteam/minecolonies#8251]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.18

### MineColonies Version

minecolonies-1.18.1-1.0.428-RELEASE

### Structurize Version

structurize-1.18.1-1.0.342-RELEASE

### Related Mods and their Versions

minecolonies-1.18.1-1.0.428-RELEASE
Forge 39.1.0
structurize-1.18.1-1.0.342-RELEASE
domum_ornamentum-1.18.1-1.0.39-ALPHA-universal

### Current Behavior

A builder got the Flu and had slowness and wasn't working but also not eating.
Firing helped and made the builder eat as suggested on the Wiki.

Switching builders switches the Flu from one character to the other.
As soon as a new builder is assigned he/she gets the Flu and it disappears from the old builder.

Also 'Slowness' keeps resetting to 30 sec. Is it ever going to pass?
Also does it matter that 'rotten flesh' was eaten. 
Except I have no idea who of the 4 eat it? :)

### Expected Behavior

Flu should stick with the character and not the Builder class itself.

### Reproduction Steps

NA

### Logs

NA

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Raycoms** [comment]
I'm 100% sure the flu doesn't pass on job switch. Can you reproduce this? Citizens can pass diseases on touch, so maybe the new builder came in and came in contact with the disease?

**Comment by Cursedth** [comment]
Sorry I have no idea how to reproduce it.
But I am sure they never touched each other and it really happened when
they changed class.

-I did however start by skipping the supply camp, thinking I had turned
it's requirement off,
 and had to go to Creative to start my Colony.
-And one person at least ate Rotten Flesh.
-I have Seren Seasons for climate and temperature, but am still in the
first Season (if applicable)

I eventually decided to kill them all and let new Citizen spawn what
resolved the issue.

Sorry, I can't be of more help.
If I have another deceased Citizen I'll try it again see if it happens
again and post some more.

Regards,
